src/index.js

```diff
@@ -1,75 +1,112 @@
-export { mix, is, like }
+'use strict';
 
-function mix(...args) {
-	var superclass = !is(args[0], 'factory') && args.shift() || baseclass
-	var factory = (is(args[args.length-1], 'factory') && args.pop()) || derive
-	superclass = is(superclass, 'mixin') ? superclass.class : derive(superclass)
-	if (args.length) factory = (org => superclass => org(args.reduce((s,m) => m.mixin(s), superclass)))(factory)
-	function mixin(superclass) {
-		var result = is(superclass, mixin) ? superclass : factory(superclass) 
-		if (mixin.classes.indexOf(result) === -1) mixin.classes.push(result)
-		return result
-	}
-	Object.defineProperties(mixin, {
-		classes: {value:[], writable:false},
-		mixins: {value:args, writable:false},
-	})
-	var Class = mixin(superclass)
-	var constructor = Class.hasOwnProperty('constructor') ? Class.constructor.bind(Class) : (...args) => new Class(...args)
-	Object.getOwnPropertyNames(Class).forEach(k => Object.defineProperty(constructor, k, {value: Class[k]}))
-	return Object.defineProperties(constructor, {
-		mixin: {value:mixin, writable:false},
-		class: {value: Class, writable:false},
-		interface: {get:(x => () => x ? x : x = getInterface(Class.prototype))()},
-	})
-}
+const baseclass = class Object {};
+const derive = superclass => ({}[superclass.name || 'Object'] = class extends superclass {});
+
+const getPropertyNames = function getPropertyNames(proto) {
+    const results = [];
+    let p = proto;
+    while (p !== Object.prototype) {
+        Object
+            .getOwnPropertyNames(proto)
+            .reduce((arr, k) => (arr.indexOf(k) === -1 ? arr.push(k) && arr : arr), results);
+
+        p = Object.getPrototypeOf(p).constructor.prototype;
+    }
+
+    return results;
+};
+
+const getInterface = function getInterface(proto) {
+    return getPropertyNames(proto).reduce((o, k) => {
+        const obj = o;
+        obj[k] = proto[k];
+        return obj;
+    }, {});
+};
 
 function is(x, type) {
-	if (typeof type == 'string') {
-		return  type == 'class'   ? is(x, 'function') && (s => /^class\s/.test(s) || /^.*classCallCheck\(/.test(s.replace(/^[^{]*{\s*/,'').replace(/\s*}[^}]*$/,'')))(x.toString()) :
-						type == 'mixin'   ? is(x, 'function') && !!x.mixin :
-						type == 'factory' ? is(x, 'function') && !is(x, 'mixin') && !is(x, 'class') && x.length == 1 :
-						typeof x == type;
-	}
-	if (typeof x == 'object') {
-		if (x instanceof type) return true
-		if (type.class && x instanceof type.class) return true
-		if (type.mixin && type.mixin.classes) return type.mixin.classes.reduce((f,c) => f || is(x,c), false)
-	}
-	else if (typeof x == 'function') {
-		if (x.mixin && x.mixin.mixins.indexOf(type) !== -1) return true
-		var c = x
-		while (c !== Object) {
-			if (c === type || c === type.class) return true
-			if (type.mixin && type.mixin.classes && type.mixin.classes.indexOf(c) !== -1) return true
-			c = c.prototype.__proto__.constructor
-		}
-	}
-	return false
-}
+    if (typeof type === 'string') {
```

Comment:
What you trade here is usually stability vs. size. A few extra bytes don't really matter, as we have internet speeds in the megabytes/second readily available in many regions and it is a lot more likely that the your user is sloppy on their end and you end up comparing a string to a number or similar.

However, I will change the triple-equal back in places which feel safe, if that's important to you.